enhancements/network/on-prem-bgp-networking.md

```diff
@@ -0,0 +1,562 @@
+
+It can be a single IPv4 or IPv6 address, or comprised of both an IPv4 and an IPv6 address, in which case BGP must advertise both addresses.
+The external `api.$cluster_name.$base-domain` DNS record should point to API VIPs.
+
+##### Ingress VIPs
+
+The Ingress VIPs are used for Ingress access and they are also provided by the user via the `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/5f7fe178c007dce006d2d59800a5685a80591ede/pkg/types/baremetal/platform.go#L194-L201)
+or `openshift-installer` terminal prompts.
+
+It can be a single IPv4 or IPv6 address, or comprised of both an IPv4 and an IPv6 address, in which case BGP must advertise both addresses.
+The external wildcard `*.apps.$cluster_name.$base-domain` DNS record should point to Ingress VIPs.
+
+
+### Highly available routed VIPs
+Access to the Kubernetes API (port 6443) and Ingress from clients both external
+and internal to the cluster should be highly available across control plane machines.
+
+The API VIPs first resides on the bootstrap instance and then move to all control plane nodes. The move will happen once the bootstrap process has completed, a FRR sidecar container residing on the boostrap will remove the VIP and FRR will update
+the leaf routing that it doesn't have the VIP anymore. 
+
+The Ingress VIPs will be created on the nodes where the Ingress Controller reports healthy.
+
+FRR will be deployed as a [static pod](https://kubernetes.io/docs/tasks/administer-cluster/static-pod/) and the assets will be rendered by the Machine Config Operator.
+When replacing a control plane machine, a new BGP session will be established with the peers. This causes no issues with the existing sessions by nature of BGP.
+
+#### Init containers
+
+- `copy-frr-static-configs`: This container will create two static files in `/etc/frr`, so FRR can be started and monitored:
+  - daemons: this file will contain the daemons that FRR will run.
+  - vtysh.conf: this file is required so the healthcheck can run (it will be used to check if FRR is running).
+- `verify-api-int-resolvable`: This container already exists for Keepalived/HAProxy and helps to ensure that the internal VIP can be resolved before creating it.
+- `render-config-frr`: This container will use baremetal-runtimecfg to render `frr.conf` based on a template that will be provided in MCO. Below we provide details on this file.
+
+#### FRR container
+
+This container will handle the start, stop and reload of BGP and BFD daemons in FRR.
+
+- it will wait for the `frr.conf` configuration file to exist (generated by baremetal-runtimecfg) and then start BGP and BFD daemons once it exists.
+- it will stop daemons if it gets a `stop` message via a socat handler running with baremetal-runtimecfg, like it's already done for HAProxy & Keepalived (and re-using the same conditions if we need to stop it for
+  feature parity). This part is already implemented for CoreDNS, HAProxy and Keepalived.
```

Comment:
Note that CoreDNS doesn't use this. It automatically reloads its config when it detects changes. If there's any way to make FRR do that we should because the socat implementation can be tricky to debug if something goes wrong.

Reply:
I wish it existed but it doesn't, the doc is clear: https://docs.frrouting.org/en/latest/setup.html#reloading
FRR has to be reloaded when a new config needs to be applied. I'll remove the coredns mention though.